modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replication/request/DualRowReplicaRequest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed.replication.request;
+
+import org.apache.ignite.internal.replicator.message.ReplicaRequest;
+import org.apache.ignite.internal.schema.BinaryRow;
+import org.apache.ignite.internal.table.distributed.replicator.action.RequestType;
+import org.apache.ignite.network.annotations.Marshallable;
+
+/**
+ * Dual row replica request.
+ */
+public interface DualRowReplicaRequest extends ReplicaRequest {
+    @Marshallable
+    BinaryRow binaryRow();
+
+    @Marshallable
+    BinaryRow oldBinaryRow();
```

Comment:
Not sure whether old is proper naming here. It's either first and second rows or SwapRowReplicaRequest with expected and new rows.

Reply:
Renamed the class name.